lib/archethic/contracts/interpreter/scope.ex

```diff
@@ -114,6 +114,32 @@ defmodule Archethic.Contracts.Interpreter.Scope do
     )
   end
 
+  def get_function_ast(function_name, nil) do
+    function_key = function_to_function_key(function_name, [])
+    get_in(Process.get(:scope), ["functions", function_key, :ast])
+  end
+
+  def get_function_ast(function_name, args) do
```

Comment:
Since this function is used only by this module you could make it private with `defp`